lib/compute/agent-node-config.ts

```diff
@@ -6,108 +6,188 @@
-      deleteRootOnTermination: true,
-      description: `jenkinsAgentNode-${config.workerLabelString}`,
-      ebsEncryptRootVolume: 'ENCRYPTED',
-      ebsOptimized: false,
-      hostKeyVerificationStrategy: 'OFF',
-      iamInstanceProfile: this.AgentNodeInstanceProfileArn,
-      idleTerminationMinutes: '30',
-      initScript: config.initScript,
-      labelString: config.workerLabelString,
-      launchTimeoutStr: '300',
-      maxTotalUses: -1,
-      minimumNumberOfInstances: 0,
-      minimumNumberOfSpareInstances: 1,
-      mode: 'EXCLUSIVE',
-      monitoring: true,
-      numExecutors: 1,
-      remoteAdmin: config.remoteUser,
-      remoteFS: '/var/jenkins',
-      securityGroups: props.agentNodeSecurityGroup,
-      stopOnTerminate: false,
-      subnetId: props.subnetId,
-      t2Unlimited: false,
-      tags: [{
-        name: 'Name',
-        value: `jenkinsAgentNode-${config.workerLabelString}`,
-      }],
-      tenancy: 'Default',
-      type: config.instanceType,
-      useEphemeralDevices: false,
-    };
-  }
-}
+     const ecrManagedPolicy = new iam.ManagedPolicy(stack, 'OpenSearch-CI-AgentNodePolicy', {
+       description: 'Jenkins agents Node Policy',
+       managedPolicyName: 'OpenSearch-CI-AgentNodePolicy',
+       statements: [
+         new iam.PolicyStatement({
+           effect: iam.Effect.ALLOW,
+           actions: [
+             "ecr-public:BatchCheckLayerAvailability",
+             "ecr-public:GetRepositoryPolicy",
+             "ecr-public:DescribeRepositories",
+             "ecr-public:DescribeRegistries",
+             "ecr-public:DescribeImages",
+             "ecr-public:DescribeImageTags",
+             "ecr-public:GetRepositoryCatalogData",
+             "ecr-public:GetRegistryCatalogData",
+             "ecr-public:InitiateLayerUpload",
+             "ecr-public:UploadLayerPart",
+             "ecr-public:CompleteLayerUpload",
+             "ecr-public:PutImage",
+           ],
+           resources: ['arn:aws:ecr-public::'+this.ACCOUNT+':repository/*'],
+           conditions: {
+            'StringEquals': {
+              'aws:RequestedRegion': this.STACKREGION,
+              'aws:PrincipalAccount': [this.ACCOUNT],
+            },
+          },
+         }),
+       ],
+       roles: [AgentNodeRole],
+     });
+     ecrManagedPolicy.addStatements(
+      new iam.PolicyStatement({
+        actions: [
+          'ecr-public:GetAuthorizationToken',
+          'sts:GetServiceBearerToken',
+        ],
+        resources: ['*'],
+        conditions: {
+          'StringEquals': {
+            'aws:RequestedRegion': this.STACKREGION,
+            'aws:PrincipalAccount': [this.ACCOUNT],
+          },
+        },
+      }),
+    );
+    if (assumeRole !== undefined) {
+      console.log('Adding agent agentAssumeRole is  👉', agentAssumeRole.valueAsString);  
+      // policy to allow assume role AssumeRole
+      AgentNodeRole.addToPolicy(
+        new iam.PolicyStatement({
```

Comment:
All the places where `iam` module is used, please correct it to first import the relevant constructs and classes and then use the same without using `iam.`